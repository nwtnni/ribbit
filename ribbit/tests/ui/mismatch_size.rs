#[ribbit::pack(size = 16)]
#[derive(Copy, Clone)]
struct A(u16);

#[ribbit::pack(size = 16)]
#[derive(Copy, Clone)]
struct B {
    #[ribbit(size = 1)]
    a: A,
    b: u15,
}

fn main() {}
