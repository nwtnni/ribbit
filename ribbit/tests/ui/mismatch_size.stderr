error[E0080]: evaluation of constant value failed
 --> tests/ui/mismatch_size.rs:8:21
  |
8 |     #[ribbit(size = 1)]
  |                     ^ the evaluated program panicked at 'Annotated size does not match actual size of type A', $DIR/tests/ui/mismatch_size.rs:8:21
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> tests/ui/mismatch_size.rs:5:1
  |
5 | #[ribbit::pack(size = 16)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | expected `u16`, found `u8`
  | arguments to this function are incorrect
  |
note: function defined here
 --> src/lib.rs
  |
  |     pub const unsafe fn unpack<T: Pack>(loose: T::Loose) -> T {
  |                         ^^^^^^
  = note: this error originates in the attribute macro `ribbit::pack` (in Nightly builds, run with -Z macro-backtrace for more info)
